simpleviewinc/tap-events-force
# Type definition review - Optional

**Ticket**: [ZEN-449](https://jira.simpleviewtools.com/browse/ZEN-449)

## Context

* Events Force has sent us a types definition file which includes some optional props
* We need to ensure all optional props are not required within the Sessions Component

## Goal

* Ensure optional props do not throw errors when not passed

## Updates

* `src/actions/attendees/initSortedAttendees.js`
  * Made `tickets` && `bookedTickets` optional
* `src/actions/attendees/setAttendeesByTicket.js`
  * Added validation helper, that only validates when the passed in prop exists
  * Otherwise sets it to an empty object
* `src/actions/session/mapSessionInterface.js`
  * Added defaults for the alert prop
* `src/mocks/eventsforce/bookingStates/index.js` && `src/mocks/eventsforce/bookingStates/optionalTickets.js`
  * Added extra booking state to test optional props

## Testing
* Run the docker package => `keg evf pack run docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-449-type-def-review`
* Navigate to http://evf-zen-449-type-def-review.local.kegdev.xyz/
* Open the testData editor, and switch to `optionalTickets-group-optionalTickets`
  * Looks like this => http://snpy.in/APSYaq
* Hit the Apply Changes button
* Open the testData editor again and inspect the test data, and ensure the `tickets` && `bookedTickets` keys do not exist
  * Looks like this => http://snpy.in/dz2cn6
* Check the browser inspector, and ensure no errors were logged
* Ensure the site works as expected
